internal/model/error.go

```diff
@@ -0,0 +1,60 @@
+package model
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+)
+
+type validationError struct {
+	message   string
+	subErrors []error
+}
+
+func newValidationError(message string) *validationError {
+	return &validationError{message: message}
+}
+
+func (v *validationError) orNil() error {
+	if len(v.subErrors) == 0 {
+		return nil
+	}
+	return v
+}
+
+func (v *validationError) Error() string {
+	if v == nil {
+		return ""
+	}
+	return strings.TrimSpace(v.errorPrefix(nil, true))
+}
+
+func (v *validationError) errorPrefix(prefix []rune, last bool) string {
+	sep := ":\n"
+	if len(v.subErrors) == 0 {
+		sep = ""
+	}
+	errMsg := bytes.NewBufferString(fmt.Sprintf("%s%s%s", string(prefix), v.message, sep))
+	for i, serr := range v.subErrors {
```

Comment:
There is some looping risk here, if the error gets added to its own subErrors tree.

Reply:
Fixed and added test